### DISCUSSION TITLE
Is it possible to draw a line annotation?

### DISCUSSION THREAD

**Discussion by SoggyBottomBoy** [Live-Charts/LiveCharts2#1586] [Q&A]
**Is your feature request related to a problem? Please describe.**
As far as I can tell a visual element needs to be of type ISizedGeometry which LineGeometry is not, and sections can only be rectangular. Neither of these options are ideal when using a strokes with a dash style as the dash returns on itself (i.e. 0 width rectangles).

I'm wondering if it is possible to draw a line as a visual element or some other element except a Series.

**Describe the solution you'd like**
Allow for 1D visual elements which are not of type ISized geometry.

**Describe alternatives you've considered**
As discussed above tried using VisualGeometry<RectangleGeometry> and RectangleSection to emulate a line but neither case was ideal.

**Additional context**

**Comment by beto-rodriguez** [comment]
Hi, I'll try to explain:

> visual element needs to be of type ISizedGeometry 

No, take a look in the docs: https://livecharts.dev/api/2.0.0-rc2/LiveChartsCore.VisualElements.VisualElement-1, especially in the "derived" section:

There you can find a lot of classes in the libraries that inherit from VisualElement, some of them are really simple, like the [GeometryVisual](https://github.com/beto-rodriguez/LiveCharts2/blob/master/src/LiveChartsCore/VisualElements/GeometryVisual.cs) (it draws basic shapes, like circles and rectangles) and some of them are more complex like the [AngularTicks](https://github.com/beto-rodriguez/LiveCharts2/blob/master/src/LiveChartsCore/VisualElements/AngularTicksVisual.cs) that draws all the thick marks in the angular gauge class.

You can use both examples as a base to build your own.

Alternatively, there is a way to use all the power of SkiaSharp to draw anything in the chart:

https://livecharts.dev/docs/WPF/2.0.0-rc2/samples.general.drawOnCanvas

I close this issue and move it to a discussion since it seems there an answer for your query, feel free to reply if you need further assitance.

**Comment by SoggyBottomBoy** [comment]
Yer I just figured wanting to draw lines would be a very common scenario. In my case there is a particular chart which indicates the point they are investigating with other charts which reflect data about this point. So in this case the cross hairs which is more to do with mouse movement isn't applicable, rather this point is fixed.

This is what I want to show:
![image](https://github.com/user-attachments/assets/0ff11d73-4a16-4473-8f7b-1289f6bc3a2d)

Right now because I am using a rectangle I have this issue with the dash style
![image](https://github.com/user-attachments/assets/ae33c532-6f5c-401a-b73c-7b637d50df0b)

But yes I can draw my own lines, I just guess I thought this would be a very common thing. We have a dialog when the user clicks the chart to export any series on that chart, so I don't really want these lines to be line series (and they don't represent a line series).

**Reply by beto-rodriguez** [reply]
This is an interesting case, I will consider this as a new feature for the library, for now sadly you need to build your own alternative, and feel free to create a PR to share your work with the community!

**Reply by SoggyBottomBoy** [reply]
Yer right now these lines are actually sections (a rectangle with no width/height). If I was to look into this, I think VisualElements should not be restricted to ISized Geometry which I expect somewhat reflects it is of 2d nature. Do you agree?

**Reply by beto-rodriguez** [reply]
The restriction you mentioned is not in the VisualElement<T>, it is in the `GeometryVisual<T>` class. It means that you can create your own visual based on the examples I provided, let me know if there is anything I can help with.

**Reply by SoggyBottomBoy** [reply]
Sorry yes, so what I therefore mean, is should the generic for GeometryVisual be restricted to ISizedGeometry. I haven't yet read enough of the code to fully understand why this constraint is imposed. I my mind I can't see why it can't be 
```csharp
public class GeometryVisual<TGeometry, TLabelGeometry, TDrawingContext> : BaseGeometryVisual<TDrawingContext>
    where TDrawingContext : DrawingContext
    where TGeometry : IGeometry<TDrawingContext>, new()
    where TLabelGeometry : ILabelGeometry<TDrawingContext>, new()
```
(TGeometry : IGeometry<TDrawingContext>)

I'll have a read through when I have the time to understand it more

**Reply by beto-rodriguez** [reply]
Another simple alternative would be to use a Rectangle that start on 0,0 and end on your target index, you can play with the ZIndex property to achieve the desired effect.
